src/RuleWithOptionsInterface.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator;
+
+/**
+ * An extended version of {@see RuleInterface} which allows exporting a rule options in addition to a rule name. It's
+ * useful for passing to frontend for further identification and implementing client-side validation. If you don't need
+ * that (for example for REST API), use {@see RuleInterface} instead.
+ */
+interface RuleWithOptionsInterface extends RuleInterface
+{
+    /**
+     * Gets a rule options as associative array. It's used for identification on the frontend with further implementing
+     * of client-side validation. Usually it's just a mapping between rule property names and values.
+     *
+     * For a example, for this rule:
+     *
+     * ```php
+     * new SomeRule(property1: 'value1', property2: 'value2');
+     * ```
+     *
+     * the options will be:
+     *
+     * ```php
+     * [
+     *     'property1' => $this->>property1,
+     *     'property2' => $this->>property2,
+     *     // ...
+     * ];
+     * ```
+     *
+     * Sometimes it's helpful to include dynamic options as well:
+     *
+     * ```php
+     * [
+     *     // ...
+     *     'dynamicOption' => $this->getDynamicOption(), // Dynamic option is not necessarily a rule class property.
+     *     // ...
+     * ];
+     * ```
+     *
+     * For messages the value is a nested array with the following structure (below is a result of the method call):
+     *
+     * ```php
+     * 'message' => [ // Array is used with no parameters too.
+     *     'template => 'A message without parameters.',
+     *     'parameters' => [], // Explicitly specified even for empty parameters.
+     * ],
+     * 'anotherMessage' => [
+     *     'template' => 'A message with {parameter}.', // Handler dependent parameters
+     *     'parameters' => ['property4' => $this->property4], // Property dependent parameters
+     * ],
+     * 'wrongMessage' => 'Wrong message.' // This is possible, but wrong. Use example above for consistent structure.
```

Comment:
`This is possible` — suggest remove this. By our agreements messages in options always must have structure:

```
'message' => ['template' => ..., 'parameters' => [...]]
```

Reply:
Fixed. Just in case, you mean remove the comment part, right? Or remove this example completely?

Reply:
Now is fine!